org.civicrm.reqangular/test/directives/time-amount-picker.directive.spec.js

```diff
@@ -0,0 +1,156 @@
+          expect(_.includes(vm.hoursOptions, 24)).toBeTruthy();
+        });
+
+        it('builds correct amount of minutes options', function () {
+          expect(vm.minutesOptions.length).toBe(60);
+        });
+
+        describe('when hours value is changed and it is less than the minimum specified', function () {
+          beforeEach(function () {
+            vm.selectedHours = 2;
+            vm.buildMinutesOptions();
+          });
+
+          it('builds correct amount of minutes options', function () {
+            expect(vm.minutesOptions.length).toBe(30);
+          });
+        });
+      });
+
+      describe('when "time-amount-picker-max-amount" attribute is passed', function () {
+        beforeEach(function () {
+          buildDirective([
+            { key: 'time-amount-picker-max-amount', value: '7.5', bind: '>' }
+          ]);
+        });
+
+        it('builds correct amount of hours options', function () {
+          expect(vm.hoursOptions.length).toBe(8);
+        });
+
+        it('has expected hours options', function () {
+          expect(_.includes(vm.hoursOptions, 0)).toBeTruthy();
+          expect(_.includes(vm.hoursOptions, 3)).toBeTruthy();
+          expect(_.includes(vm.hoursOptions, 7)).toBeTruthy();
+        });
+
+        it('builds correct amount of minutes options', function () {
+          expect(vm.minutesOptions.length).toBe(60);
+        });
+
+        describe('when hours value is changed and it is same as the maximum specified', function () {
+          beforeEach(function () {
+            vm.selectedHours = 7;
+            vm.buildMinutesOptions();
+          });
+
+          it('builds correct amount of minutes options', function () {
+            expect(vm.minutesOptions.length).toBe(31);
+          });
+        });
+      });
+
+      describe('when "time-amount-picker-value" attribute is passed', function () {
+        beforeEach(function () {
+          buildDirective([
+            { key: 'time-amount-picker-value', value: '6.5', bind: '=' }
+          ]);
+        });
+
+        it('sets selected hours', function () {
+          expect(vm.selectedHours).toBe('6');
+        });
+
+        it('sets selected minutes', function () {
+          expect(vm.selectedMinutes).toBe('30');
+        });
+
+        describe('when selected values are changed', function () {
+          beforeEach(function () {
+            vm.selectedHours = 8;
+            vm.selectedMinutes = 45;
+
+            vm.calculateSelectedValue();
+          });
+
+          it('sets the result value back', function () {
+            expect($scope.$$childTail.timeAmountPickerValue).toBe(8.75);
+          });
+        });
+      });
+    });
+
+    /**
+     * Builds a directive
+     *
+     * @param {Array} options objects of attributes keys, values and binding types
+     *   [{ key: 'key', value: 'value', bind: '<|@' }, {...}, ...]
+     */
+    function buildDirective (options) {
```

Comment:
In the future we might want to make a helper out of this `buildDirective` :)

Reply:
Ah, you like it! 😍